iree/compiler/InputConversion/MHLO/test/convert_mhlo_to_linalg_ext.mlir

```diff
@@ -30,16 +31,17 @@ func @sort_with_cst(%arg0: tensor<1x10xi32>) -> tensor<1x10xi32> {
   return %1 : tensor<1x10xi32>
 }
 
-// CHECK-LABEL: func @sort_with_cst
-// CHECK:         %[[ARG0:[a-zA-Z0-9]+]]
+// CHECK-LABEL: func @sort_with_cst(
+// CHECK-SAME:      %[[ARG0:[a-zA-Z0-9]+]]
+// CHECK-SAME:  )
 // CHECK:         %[[CST:.+]] = mhlo.constant dense<0> : tensor<i32>
-// CHECK:         %{{.+}} = linalg_ext.sort dimension(1) outs(%[[ARG0]] : tensor<1x10xi32>)  {
+// CHECK:         %[[SORT:.+]] = linalg_ext.sort dimension(1) outs(%[[ARG0]] : tensor<1x10xi32>)  {
 // CHECK:         ^bb0(%[[ARG1:.+]]: i32, %{{.*}}: i32)
-// CHECK:           %[[SCALAR:.+]] = tensor.extract %[[CST]][] : tensor<i32>
+// CHECK:           %[[SCALAR:.+]] = builtin.unrealized_conversion_cast %[[CST]] : tensor<i32> to i32
```

Comment:
The unrealized conversions should go away once the rest of mhlo is lowered. This is only applied for tensors-that-are-really-scalar, which don't survive past MHLO (and shouldn't end up bufferized). This wouldn't support just silently going from a cast of tensor->scalar to one of memref->scalar. Someone would have to very specifically implement a lowering of unrealized_conversion_cast for that.

Another option we have is to do the lowering of MHLO to linalg and linalg_ext all in one pass by including the patterns ourselves rather than running them in two separate passes. We could also add additional validation passes throughout the pipeline that IR has only the set of expected dialects (to avoid having to bundle everything into one pass, but still get the validation)

Reply:
I see I see, thanks! My concern was unrealized_conversion_cast might lives until late passes. If it's not the case, it looks good to me.

Lowering MHLO to linalg and linalg_ext all in one pass sounds good to me. Linalg and LinalgExt could get merged at some point in my mind. We could also prioritize the patterns we have in case that there are multiple patterns for lowering an op. We don't have to do it in the PR. The PR LGTM.